core/src/Cuda/Kokkos_Cuda_Parallel.hpp

```diff
@@ -1822,15 +1822,15 @@ class ParallelReduce<FunctorType, Kokkos::TeamPolicy<Properties...>,
   }
 
   inline void execute() {
-    const int nwork            = m_league_size * m_team_size;
+    const bool do_work         = m_league_size > 0 && m_team_size > 0;
```

Comment:
Did you consider
```suggestion
    const typename Policy::size_type nwork = m_league_size * m_team_size;
```
?

Reply:
That doesn't work because the product is still done with `int`s. At assignment the product is already wrong.
We could cast one of the operands on the right-hand side but that might still not be enough, see https://github.com/kokkos/kokkos/issues/4531#issuecomment-970595549.
In the end, it is much better to just avoid doing these multiplications if we can.